src/docgen/transpile/transpile.ts

```diff
@@ -642,36 +646,48 @@ export abstract class TranspileBase implements Transpile {
   }
 
   protected findSubmodule(type: reflect.Type): reflect.Submodule | undefined {
-    if (!type.namespace) {
-      return undefined;
+
+    if (this.submodulesCache.has(type.fqn)) {
+      return this.submodulesCache.get(type.fqn);
     }
 
-    // if the type is in a submodule, the submodule name is the first
-    // part of the namespace. we construct the full submodule fqn and search for it.
-    const submoduleFqn = `${type.assembly.name}.${type.namespace.split('.')[0]}`;
-    const submodules = type.assembly.submodules.filter(
-      (s) => s.fqn === submoduleFqn,
-    );
+    let submodule = undefined;
+    if (type.namespace) {
+      // if the type is in a submodule, the submodule name is the first
+      // part of the namespace. we construct the full submodule fqn and search for it.
+      const submoduleFqn = `${type.assembly.name}.${type.namespace.split('.')[0]}`;
+      const submodules = type.assembly.submodules.filter(
```

Comment:
According to the flame graph, the `assembly.submodules` call is also pretty expensive. The submodules cache here reduced another 15 seconds from the render time.